libcextract/SymbolExternalizer.cpp

```diff
@@ -732,28 +687,54 @@ std::string SymbolExternalizer::Get_Modifications_To_Main_File(void)
   return std::string(main_buf.begin(), main_buf.end());
 }
 
-void SymbolExternalizer::Strongly_Externalize_Symbol(const std::string &to_externalize)
-{
-  _Externalize_Symbol(to_externalize, ExternalizationType::STRONG);
-}
-
-void SymbolExternalizer::Weakly_Externalize_Symbol(const std::string &to_externalize)
-{
-  _Externalize_Symbol(to_externalize, ExternalizationType::WEAK);
-}
-
 /** Given a MacroExpansion object, we try to get the location of where the token
     appears on it.
 
     TODO: clang may provide a way of doing this with a tokenizer, so maybe this
     code can become cleaner with it.  */
-static std::vector<SourceRange>
-Get_Range_Of_Identifier_In_Macro_Expansion(const MacroExpansion *exp, const char *id)
+std::vector<std::pair<std::string, SourceRange>>
+SymbolExternalizer::Get_Range_Of_Identifier_In_Macro_Expansion(const MacroExpansion *exp)
 {
-  return Get_Range_Of_Identifier_In_SrcRange(exp->getSourceRange(), id);
+  std::vector< std::pair < std::string, SourceRange> > ret;
```

Comment:
there is no need to construct this std::pair if we are going to access the SymbolUpdateStatus map. Just return the interesting ranges.